eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/UnifiedSet.java

```diff
@@ -1282,8 +1282,24 @@ public boolean equals(Object object)
             return false;
         }
 
-        Set<?> other = (Set<?>) object;
-        return this.size() == other.size() && this.containsAll(other);
+        Set<?> s = (Set<?>) object;
+        if (s.size() != this.size())
+        {
+            return false;
+        }
+
+        try
+        {
+            return this.containsAll(s);
+        }
+        catch (ClassCastException ignored)
+        {
+            return false;
+        }
+        catch (NullPointerException ignored)
+        {
+            return false;
+        }
```

Comment:
Why change this implementation? How did it come up?

Reply:
I homogenized it with the others implementation of `Set` where this happens, for example in `java.util.AbstractSet`. I think it is necessary in order to be coherent with the public contract of `java.util.Collection.containsAll(Collection<?>)` which can throw this exception.

Reply:
Are the `return false` lines covered? If not, I think we should revert this change. It's unrelated and in one of the most heavily used classes.

Reply:
@motlin ok